include/boost/dll/shared_library.hpp

```diff
@@ -177,7 +177,7 @@ class shared_library
     *
     * \throw Nothing.
     */
-    ~shared_library() BOOST_NOEXCEPT {}
+    virtual ~shared_library() BOOST_NOEXCEPT {}
 
```

Comment:
Ok, that actually makes sense, especially since smart_lib is not abstract.

I actually added the public get_void to the shared_lib, which is all I need - so using shared_lib as a member is not a problem. Wouldn't that be alright with this member-function?